core.ts

```diff
@@ -200,845 +200,844 @@ class Poll<
-		}
-		if (Array.isArray(filter)) {
-			return filter;
-		}
-		return [`${filter}`];
-	};
+	return encodedResource;
+};
 
-	const applyBinds = (
-		filter: string,
-		params: { [index: string]: PinejsClientCoreFactory.Filter },
-		parentKey?: string[],
-	) => {
-		for (const index in params) {
-			const param = params[index];
-			let paramStr = `(${buildFilter(param).join('')})`;
-			// Escape $ for filter.replace
-			paramStr = paramStr.replace(/\$/g, '$$$$');
-			filter = filter.replace(
-				new RegExp(`\\$${index}([^a-zA-Z0-9]|$)`, 'g'),
-				`${paramStr}$1`,
-			);
-		}
-		filter = `(${filter})`;
-		return addParentKey(filter, parentKey);
-	};
+const escapeCountableResource = (resource: string | string[]) => {
+	return baseEscapeResource(resource).replace(trailingCountRegex, '/$count');
+};
 
-	const filterOperation = (
-		filter: PinejsClientCoreFactory.FilterOperationValue,
-		operator: PinejsClientCoreFactory.FilterOperationKey,
-		parentKey?: string[],
-	) => {
-		const op = ' ' + operator.slice(1) + ' ';
-		if (isPrimitive(filter)) {
-			const filterStr = escapeValue(filter);
-			return addParentKey(filterStr, parentKey, op);
-		} else if (Array.isArray(filter)) {
-			const filterArr = handleFilterArray(filter);
-			const filterStr = bracketJoin(filterArr, op);
-			return addParentKey(filterStr, parentKey);
-		} else if (isObject(filter)) {
-			const result = handleFilterObject(filter);
-			if (result.length < 1) {
-				throw new Error(
-					`${operator} objects must have at least 1 property, got: ${JSON.stringify(
-						filter,
-					)}`,
-				);
+// Escape a primitive value
+const escapeValue = (value: null | string | number | boolean | Date) => {
+	if (isString(value)) {
+		value = value.replace(/'/g, "''");
+		return `'${encodeURIComponent(value)}'`;
+	} else if (isDate(value)) {
+		return `datetime'${value.toISOString()}'`;
+	} else if (value === null || NumberIsFinite(value) || isBoolean(value)) {
+		return value;
+	} else {
+		throw new Error('Not a valid value: ' + typeof value);
+	}
+};
+
+const join = (strOrArray: string | string[], separator = ',') => {
+	if (isString(strOrArray)) {
+		return strOrArray;
+	} else if (Array.isArray(strOrArray)) {
+		return strOrArray.join(separator);
+	} else {
+		throw new Error('Expected a string or array, got: ' + typeof strOrArray);
+	}
+};
+
+// Join together a bunch of statements making sure the whole lot is correctly parenthesised
+const bracketJoin = (arr: string[][], separator: string) => {
+	if (arr.length === 1) {
+		return arr[0];
+	}
+	const resultArr: string[] = [];
+	arr
+		.map(subArr => {
+			if (subArr.length > 1) {
+				return `(${subArr.join('')})`;
 			}
-			if (result.length === 1) {
-				return addParentKey(result[0], parentKey, op);
+			return subArr[0];
+		})
+		.forEach((str, i) => {
+			if (i !== 0) {
+				resultArr.push(separator);
+			}
+			resultArr.push(str);
+		});
+	return resultArr;
+};
+
+// Add the parentKey + operator if (it exists.) {
```

Comment:
Weirdly formatted comment

Reply:
Yeah, that'll have been whilst converting from coffeescript...